rest/server_context.go

```diff
@@ -514,15 +523,24 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(ctx context.Context, config
 				if err := base.WaitUntilDataStoreExists(dataStore); err != nil {
 					return nil, fmt.Errorf("attempting to create/update database with a scope/collection that is not found")
 				}
-
-				if err := initDataStore(dataStore); err != nil {
+				metadataIndexOption := db.IndexesWithoutMetadata
+				if base.IsDefaultCollection(scopeName, collectionName) {
+					hasDefaultCollection = true
+					metadataIndexOption = db.IndexesAll
+				}
+				if err := initDataStore(dataStore, metadataIndexOption); err != nil {
 					return nil, err
 				}
 			}
 		}
+		if !hasDefaultCollection {
+			if err := initDataStore(bucket.DefaultDataStore(), db.IndexesMetadataOnly); err != nil {
```

Comment:
As mentioned in Slack - don't think this is fully working. The metadata-only default collection gets a full set of indexes.